mrdo_arcade.asm

```diff
@@ -8333,14 +8344,52 @@ LOC_B8B1:
 	LD		HL, 78H
 	CALL	REQUEST_SIGNAL
 	JP		LOC_D36D
-LOC_B8EC:
+LOC_B8EC: ; Enter Alpha monster chomper mode
 	LD		A, 80H
 	LD		($72C3), A
+
+	; Check Bit 6 of the ball state
+	; Bit 6 is set to 1 when the ball is in the cooldown phase
+	; We only want to return the ball during this interval
+
+	LD      IY, $72D9         ; Load ball state pointer
+	BIT     6, (IY+0)         ; Check if bit 6 is set
+	JR      NZ, SKIP_BALL_RETURN ; Skip if not in cooldown phase
+
+  ; Return the ball
+	LD      A, 20H            ; Set only bit 5 (00100000)
+	LD      (IY+0), A         ; Store initial state

```

Comment:
This sequence is overwriting the whole value of (IY+0).
If you want to set bit 5 only, you can use the SET instruction and replace these two lines by

SET 5,(IY+0)